packages/server/api.js

```diff
@@ -234,6 +234,56 @@ module.exports = async function attachAPI(app, {wss, db, dbDir}) {
     ])
   }
 
+  app.post('/api/emotes', [
+    ...middleware.loadSessionID('sessionID'),
+    ...middleware.getSessionUserFromID('sessionID', 'sessionUser'),
+    ...middleware.requireBeAdmin('sessionUser'),
+    ...middleware.loadVarFromBody('shortcode'),
+    ...middleware.validateVar('shortcode', validate.string),
+    ...middleware.requireNameValid('shortcode'),
+    ...middleware.loadVarFromBody('imageURL'),
+    ...middleware.validateVar('imageURL', validate.string),
+
+    async function(request, response, next) {
+      const { imageURL, shortcode } = request[middleware.vars]
+
+      if (await db.emotes.findOne({shortcode})) {
+        response.status(400).json(errors.NAME_ALREADY_TAKEN)
+        return
+      }
+
+      const newEmote = {imageURL, shortcode}
+      await db.emotes.insert(newEmote)
+      sendToAllSockets('emote/new', {
+        emote: await serialize.emote(newEmote)
+      })
+      response.status(200).json({})
```

Comment:
`201 Created` please. Same goes for `POST /api/users` and friends.

Reply:
👍